src/main/java/Parser.java

```diff
@@ -5,83 +5,118 @@
  * Bridges communication between the user and the tasks
  */
 public class Parser {
+    public static boolean isRunning = true;
+    private static TaskList tasklist;
+    private static String command;
+    private static String[] commandArr;
     /**
      * Handles user input regarding tasks
-     * @param command String that users enter
+     *
+     * @param userString String that users enter
      */
-    public static void handle(String command, TaskList tasklist) throws DukeException {
-        String[] arr = command.split(" ");
+    public static void handle(String userString, TaskList tasklist) throws DukeException {
+        Parser.command = userString;
+        Parser.tasklist = tasklist;
+        String[] arr = userString.split(" ");
         String firstWord = arr[0];
+        Parser.commandArr = arr;
 
         switch (firstWord) {
-        case "done":
-            if (arr.length < 2) {
-                throw new MissingTaskNumberException("Missing task number");
-            }
-            int doneTaskNumber = Integer.parseInt(arr[1]);
-            try {
-                tasklist.get(doneTaskNumber).setDone(true);
-            } catch (IndexOutOfBoundsException e) {
-                throw new InvalidTaskNumberException("Task does not exist");
-            }
-            System.out.printf("Solid work! I've marked task %d as done.%n", doneTaskNumber);
+        case "bye":
+            UI.bye();
+            isRunning = true;
+            break;
+        case "list":
             tasklist.list();
             break;
+        case "done":
+            handleDone();
+            break;
         case "todo":
-            if (arr.length < 2) {
-                throw new MissingTaskNameException("Missing task name");
-            }
-            String remaining = command.substring(5);
-            tasklist.add(new ToDo(remaining));
-            System.out.println("Great! I've added your todo. Enter 'list' to see your tasks!");
-            System.out.printf("You currently have %d tasks.%n", tasklist.size());
+            handleTodo();
```

Comment:
I like you encasuplating all these codes into a method to make your code simplier!

Reply:
Thanks! 👍